### ISSUE TITLE
Overriding with bomProperty does not work when applying spring-boot plugin in a single module

### ISSUE THREAD

**Issue by jfourment** [spring-gradle-plugins/dependency-management-plugin#219]
Hi, 

We are using dependency management with spring boot on multi module application.
One of this module contains the main class and some properties and aggregates all other modules to build a spring boot jar:
- moduleA
- moduleB
- module-jar (which depends on A and B) 

Dependency management is defined in root build.gradle for all subprojects and imports spring-boot-dependencies bom (we are on 1.5.x at the moment)

The plan is to use spring-boot plugin in conjuction with dependency management plugin. In order to have dependency management for all modules and spring boot app built in module-jar, spring boot plugin is not applied by default, and is applied locally in module-jar.

Root build.gradle looks like this:
```
plugins {
    id "io.spring.dependency-management" version "1.0.5.RELEASE"
    id "org.springframework.boot" version "1.5.14.RELEASE" apply false
}
subprojects {
    apply plugin: 'io.spring.dependency-management'
    configurations {
        all {
            dependencyManagement {
                imports {
                    mavenBom ("org.springframework.boot:spring-boot-dependencies:1.5.14.RELEASE") {
                            bomProperty "hibernate.version", "5.2.17.Final"
                    }
                }
```

And for module-jar:
```
apply plugin: 'org.springframework.boot'
dependencies {
    implementation group: 'org.hibernate', name: 'hibernate-core'
....
}
```

Now the problem is, that by doing that the override I did on the import using ```bomProperty``` is ignored, and I don't understand why... It seems somewhat related to the plugin being applied in module-jar (If i don't apply the plugin, dependencies are correctly evaluated)

Overriding with ```ext['hibernate.version'] = blah``` in buildscript seems to work though. 

What am I doing wrong?

Note: we are using gradle 4.8.1.

**Comment by wilkinsona** [comment]
You're not really doing anything wrong aside from using Gradle 4 which isn't officially supported with Boot 1.5.

Applying Boot's plugin causes it to import the `spring-boot-dependencies` bom. This import happens after your own import and, when multiple overlapping boms are imported, [the last one wins](https://github.com/spring-gradle-plugins/dependency-management-plugin#importing-multiple-boms). Setting `ext['hibernate.version'] = '5.2.17.Final'` works as it's effectively a global bom property that applies to both your import of the bom and the Spring Boot plugin's import of the bom.

In short, this plugin is behaving as designed and documented with a later import of a bom overriding and earlier one. I'd recommend using `ext['hibernate.version'] = '5.2.17.Final'`.

For the behaviour to be different, changes both in Spring Boot and in this plugin would be necessary. This plugin would have to provide a mechanism for importing a bom only if it hasn't already been imported and Spring Boot would have to make use of the mechanism. If you'd like to see that enhancement, please open a Spring Boot issue for our consideration and we can take things from there.

**Comment by jfourment** [comment]
Thank you for the explanation. It's a lot clearer now. 
I guess it would be a nice to have feature, unless I'm the only one here trying to pull that kind of stunt. I will open a request.

As for the Boot 1.5 support , I didn't realize it was not supported at all, but definitely felt the pain of not having Boot 2.x. A migration is on the cards though.
